HTML structural change + invertTransparency option...

agamemnus opened the pull request:
1) Now the HTML is structured like this:
````
<div class="wrapper">
 <wave>
  <wave><canvas/></wave> <--- regular wave
  <wave><canvas/></wave> <--- progress wave
 <wave>
</div>
````
Versus the old way:
````
<div class="wrapper">
 <wave>
  <wave><canvas/></wave> <--- progress wave
  <canvas/> <--- regular wave
 <wave>
</div>
````
2) I also added an `invertTransparency` option. It can be used like this: `invertTransparency: true` or `invertTransparency: {cutColor: 'black'}`
This will create a transparent area where the wave normally is. The `cutColor` (default `#fefefe`) slightly affects how the wave is cut out -- how crisp it is.

3) Miscellaneous structural changes...

4) Added `cursorAlignment` option. Default is "middle" (before this change, cursors always aligned to the left). Can be one of `left`, `middle`, or `right`. Anything other than `middle` or `right` is assumed to be `left`.

agamemnus commented:
Not sure how to fix this one:
````
   src/drawer.multicanvas.js
    294 |                ['wave'].concat(canvas.progressWaveCtx ? ['progressWave'] : []).forEach (function (waveType) {
                                                                                                  ^ Don't make functions within a loop.
````

agamemnus commented:
Argle bargle! It doesn't want me to put a forEach loop inside another forEach loop!!!

thijstriemstra commented:
@agamemnus see https://travis-ci.org/katspaugh/wavesurfer.js/builds/241874764#L333

agamemnus commented:
Oops.

thijstriemstra commented:
You can add `294` to .jshintrc in the root to ignore that forEach error.

agamemnus commented:
I'm using the web ui...

thijstriemstra commented:
That doesn't matter. Look up the files in your `patch-18` branch using the web ui.

thijstriemstra commented:
No I'm suggesting to add an additional line there.

thijstriemstra commented:
But to be honest, you should learn how to fork this project and make changes from your development machine. That way you can run the tests yourself before committing a change.

thijstriemstra commented:
And before you continue, this is a pretty big change that probably needs to be introduced in v2 @katspaugh?

agamemnus commented:
I tried both false and true and it didn't make a difference.

katspaugh commented:
I should probably merge the next branch into master asap.

agamemnus commented:
A few more small changes.

* I added styleList and classList as parameters. Attributes currently are `progressWave` and `wave`. With this, you can insert classes (a string) or specific styles (an object).

* Removed some duplicate code.

(The `Canvas` Drawer has not yet been updated, pending a review of my changes to the `MultiCanvas` Drawer.)

agamemnus commented:
For competitive reasons, I can't post a screenshot of my wave design. However, I would like to provide a possible utility for setting the classes of the wave and progressWave: one may choose to place a border-bottom of a different color for each, thus creating a secondary way to show users the audio progress.

agamemnus commented:
Not sure what is going on now.

thijstriemstra commented:
Please break this PR into smaller changes/PRs.

agamemnus commented:
OK.
